streamlit-python/src/module-ui/src/ui.py

```diff
@@ -1,24 +1,49 @@
-# Import python packages
-import streamlit as st
-from snowflake.snowpark.context import get_active_session
-
-# Write directly to the app
-st.title("Hello Snowflake!")
-st.write(
-   """The sum of the two numbers is calculated by the Python add_fn() function
-      which is called from core.add() UDF defined in your setup_script.sql.
-   """)
-
-num1 = st.number_input('First number', value=1)
-num2 = st.number_input('Second number', value=1)
-# Get the current credentials
-session = get_active_session()
-
-#  Create an example data frame
-data_frame = session.sql("SELECT core.add(%s, %s);" % (num1, num2))
-
-# Execute the query and convert it into a Pandas data frame
-queried_data = data_frame.to_pandas()
-
-# Display the Pandas data frame as a Streamlit data frame.
-st.dataframe(queried_data, use_container_width=True)
\ No newline at end of file
+def run_streamlit():
+   # Import python packages
+   # Streamlit testing framework requires imports to reside here
```

Comment:
Rather than say what's necessary here can we link to the official docs which show this to be the case? It would also help users discover those docs and learn more about the testing framework.

Alternatively, add the reason why imports can't be at the top-level to aid in the user's understanding.

Reply:
I don't know the reason - All I got is that this is by design. I will see if the official documentation mentions this, and I can link it.